### PR TITLE
[PFS][Part2] Add upsert method

### PR DESCRIPTION
**Pull request by roopakv**
# Description
Please include a summary of the change and which issue is fixed. Include the motivation for the changes, and comment on your PR if necessary for clarity.

Fixes: Part of https://app.asana.com/0/1141250355664307/1144434411596969

# Type of change

- [ ] Bug fix (change which fixes an issue)
- [x] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [x] Successfully tested locally (or docker image)
- [ ] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] New and existing unit tests pass locally with my changes
- [x] I have created or modified unit tests to sufficiently cover my changes.

### CODE REVIEW

File: Model/FeatureSwitchRepository.php [hunk 1]
```diff
@@ -26,6 +26,30 @@ public function getByName($name) {
...
+     * @param boolean $value
+     * @param boolean $defaultValue
+     * @param int $rolloutPercentage
+     * @return \Bolt\Boltpay\Api\Data\FeatureSwitchInterface
+     */
+    public function upsertByName($name, $value, $defaultValue, $rolloutPercentage) {
+        $switch = null;
+        try {
+            $switch = $this->getByName($name);
+        } catch(\Exception $e) {
```
Comment (daisy1754) [inline Model/FeatureSwitchRepository.php]: can we be more specific and have NoSuchEntityException here?

File: Model/FeatureSwitchRepository.php [hunk 2]
```diff
@@ -26,6 +26,30 @@ public function getByName($name) {
...
+            $switch ->setName($name);
+        }
+        $switch ->setValue($value);
+        $switch ->setDefaultValue($defaultValue);
+        $switch ->setRolloutPercentage($rolloutPercentage);
+        return $this->save($switch);
+    }
+
     public function save(\Bolt\Boltpay\Api\Data\FeatureSwitchInterface $switch) {
         $switch->getResource()->save($switch);
```
Comment (YevhenShBolt) [inline Model/FeatureSwitchRepository.php]: Does it need to catch some exceptions here?

Comment (roopakv) [reply Model/FeatureSwitchRepository.php]: i was considering make it such that the method calling this should catch? lmk what you think

File: Model/FeatureSwitchRepository.php (hunk 1 above)
Comment (roopakv) [reply Model/FeatureSwitchRepository.php]: sure

**Pull request merged**